network/src/logging.rs

```diff
@@ -72,6 +73,18 @@ impl<'a> NetworkSchema<'a> {
             .network_address(&metadata.addr)
     }
 
+    // Increments a counter, and logs with the network context.
+    pub fn with_count(self, counter: &IntCounterVec, amount: i64) -> Self {
+        counter
+            .with_label_values(&[
+                self.network_context.role().as_str(),
+                self.network_context.network_id().as_str(),
+                self.network_context.peer_id().short_str().as_str(),
+            ])
+            .inc_by(amount);
+        self
+    }
+
```

Comment:
This is an interesting idea :)

Reply:
After all the requests from others, I was seeing if we could automate the count dimensions in this way.  It's a little harder to have it not custom made, at least in this way.  If there are no dimensions, or at least static dimensions, they're probably pretty easy to set.